Library/Formula/apache-spark.rb

```diff
@@ -18,6 +25,9 @@ def install
   end
 
   test do
-    system "#{bin}/spark-shell <<<'sc.parallelize(1 to 1000).count()'"
+    # spark-shell has a REPL-history type of thing but it needs to write 
+    # to ~/.spark_history... not necessarily sandbox-friendly! But --driver-java-options
+    # lets us pass JVM flags like -D to let us override Java/Scala system properties!
+    system "#{bin}/spark-shell --driver-java-options '-Duser.home=/tmp/' <<<'sc.parallelize(1 to 1000).count()'"
```

Comment:
Just use `ENV.java_cache` here. It sets the user home. This is a common enough problem that it has an official Homebrew workaround :tada: